File: android/auto-mobile-sdk/src/main/kotlin/dev/jasonpearson/automobile/sdk/NavigationListener.kt
package dev.jasonpearson.automobile.sdk

/**
 * Listener interface for receiving navigation events.
 */
fun interface NavigationListener {
    /**
     * Called when a navigation event occurs.
     *
     * @param event The navigation event that occurred
     */
    fun onNavigationEvent(event: NavigationEvent)
}

File: ios/auto-mobile-sdk/Sources/AutoMobileSDK/Navigation/NavigationListener.swift
import Foundation

/// Functional interface for receiving navigation events.
public protocol NavigationListener: AnyObject, Sendable {
    func onNavigationEvent(_ event: NavigationEvent)
}

/// Closure-based navigation listener for convenience.
public final class BlockNavigationListener: NavigationListener, @unchecked Sendable {
    private let block: @Sendable (NavigationEvent) -> Void

    public init(_ block: @escaping @Sendable (NavigationEvent) -> Void) {
        self.block = block
    }

    public func onNavigationEvent(_ event: NavigationEvent) {
        block(event)
    }
}
